Add recommendations

Grmelik opened the pull request.

Grmelik committed:
Add recommendations

Grmelik committed:
Add recommendations

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/storage/FilmStorage.java

Grmelik committed:
Add recommendations

Grmelik committed:
Add recommendations

Grmelik committed:
Add recommendations

Grmelik committed:
Add recommendations

InstaYangire commented on src/test/java/ru/yandex/practicum/filmorate/service/FilmServiceTest.java:
```diff
@@ -21,6 +23,7 @@
 import static org.junit.jupiter.api.Assertions.*;
 
 class FilmServiceTest {
+    private static final Logger log = LoggerFactory.getLogger(FilmServiceTest.class);
```
Зачем тут добавлен логгер - что сюда строкой что импортом - у тебя далее в этом классе никаких изменений. У тебя написан отдельный класс под тесты. А тут пустой ненужный логгер.

InstaYangire commented on src/main/java/ru/yandex/practicum/filmorate/storage/film/InMemoryFilmStorage.java:
Зачем сложные изменения этого класса пошли - при том что тесты потом по итогу пошли Mokito?

InstaYangire commented on src/main/java/ru/yandex/practicum/filmorate/storage/film/FilmDbStorage.java:
```diff
@@ -257,4 +257,37 @@ public List<Film> searchFilms(String query, List<String> by) {
     }
+
+    // Getting recommendations
+    public List<Film> getRecommendations(int userId) {
```
Не вижу никакого теста проверяющего корректность этого метода

InstaYangire committed:
Реализация рекомендаций

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/storage/FilmStorage.java

Grmelik committed:
Add recommendations

Grmelik committed:
Merge branch 'develop' into add-recommendations

Grmelik committed:
Add recommendations

Grmelik committed:
Add recommendations

Grmelik committed:
Add recommendations
